openedx/features/course_experience/utils.py

```diff
@@ -182,23 +94,6 @@ def recurse_mark_auth_denial(block):
     course_key = CourseKey.from_string(course_id)
     course_usage_key = modulestore().make_course_usage_key(course_key)
 
-    # Deeper query for course tree traversing/marking complete
-    # and last completed block
-    block_types_filter = [
```

Comment:
Is this just to ensure we see all potential blocks?

Reply:
Correct, without this we won't have the completion of custom blocks. I suppose that `RESUME_BLOCK` could be bugged too in such case, but I didn't test it.

Reply:
The `resume_block` attribute is set along with completion in the BlockCompletionTransformer so those will go hand-in-hand.

Reply:
@Dillon-Dumesnil, exactly, I was just mentioning the scenario with `block_types_filter`. Using `BlockCompletionTransformer` resolves both issues.